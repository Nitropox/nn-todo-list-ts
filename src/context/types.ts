import { AddTaskAction, RemoveTaskAction, MakeTaskDoneAction } from "./actions";
export interface Task {
  date: number;
  title: string;
  checked: boolean;
}

export enum ActionTypes {
  addTask,
  removeTask,
  makeTaskDone
}

export type Action = AddTaskAction | RemoveTaskAction | MakeTaskDoneAction;
