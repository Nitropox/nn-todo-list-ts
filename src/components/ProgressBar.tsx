import React from "react";

export const ProgressBar: React.FC = () => {
  return <div></div>;
};
