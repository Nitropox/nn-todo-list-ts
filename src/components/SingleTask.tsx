import React from "react";

export const SingleTask: React.FC = () => {
  return <div></div>;
};
