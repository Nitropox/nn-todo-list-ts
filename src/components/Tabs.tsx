/* eslint-disable jsx-a11y/anchor-is-valid */
import React from "react";

interface TabsProps {
  activeTab: number;
  switchTab: Function;
}

export const Tabs: React.FC<TabsProps> = ({ activeTab, switchTab }) => {
  return (
    <ul className="nav nav-tabs">
      <li className="nav-item">
        <a
          className={`nav-link ${activeTab === 0 ? " active" : null}`}
          href="#"
          onClick={() => switchTab(0)}
        >
          Wszystkie
        </a>
      </li>
      <li className="nav-item">
        <a
          className={`nav-link ${activeTab === 1 ? " active" : null}`}
          href="#"
          onClick={() => switchTab(1)}
        >
          Wykonane
        </a>
      </li>
      <li className="nav-item">
        <a
          className={`nav-link ${activeTab === 2 ? " active" : null}`}
          href="#"
          onClick={() => switchTab(2)}
        >
          Niewykonane
        </a>
      </li>
    </ul>
  );
};
