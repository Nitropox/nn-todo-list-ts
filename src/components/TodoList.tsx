import React from "react";

export const TodoList: React.FC = () => {
  return <div></div>;
};
