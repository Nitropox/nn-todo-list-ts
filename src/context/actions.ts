import { ActionTypes, Task } from "./types";

type Dispatch<A> = (value: A) => void;

export interface AddTaskAction {
  type: ActionTypes.addTask;
  payload: Task;
}

export interface RemoveTaskAction {
  type: ActionTypes.removeTask;
  payload: number;
}

export interface MakeTaskDoneAction {
  type: ActionTypes.makeTaskDone;
  payload: number;
}

export const addTask = (dispatch: Dispatch<AddTaskAction>) => (
  date: number,
  title: string,
  callback?: Function
) => {
  dispatch({
    type: ActionTypes.addTask,
    payload: { date, title, checked: false }
  });
  if (callback) {
    callback();
  }
};

export const removeTask = (dispatch: Dispatch<RemoveTaskAction>) => (
  date: number
) => {
  dispatch({ type: ActionTypes.removeTask, payload: date });
};

export const makeTaskDone = (dispatch: Dispatch<MakeTaskDoneAction>) => (
  date: number
) => {
  dispatch({ type: ActionTypes.makeTaskDone, payload: date });
};
