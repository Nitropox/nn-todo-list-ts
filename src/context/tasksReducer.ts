import createDataContext from "./createDataContext";
import { ActionTypes, Task, Action } from "./types";
import { addTask, removeTask, makeTaskDone } from "./actions";
const todosReducer = (state: Task[], action: Action) => {
  switch (action.type) {
    case ActionTypes.addTask:
      return [...state, action.payload];
    case ActionTypes.removeTask:
      return state.filter(task => task.date !== action.payload);
    case ActionTypes.makeTaskDone:
      return state.map(task =>
        task.date === action.payload
          ? { ...task, checked: !task.checked }
          : task
      );
    default:
      return state;
  }
};

export const { Context, Provider } = createDataContext(
  todosReducer,
  {
    addTask,
    removeTask,
    makeTaskDone
  },
  [
    {
      date: 1570897439971,
      title: "Wykonać zadanie rekrutacyjne dla NN",
      checked: false
    },
    {
      date: 1570897439972,
      title: "Sprawdzić moliwości w świecie JS",
      checked: false
    },
    {
      date: 1570897439973,
      title: "Ukończyć kurs React Hooks",
      checked: true
    }
  ]
);
