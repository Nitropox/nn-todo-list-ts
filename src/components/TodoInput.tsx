import React, { useState, useContext } from "react";
import { Context as TodosContext } from "../context/tasksReducer";

export const TodoInput = () => {
  const [input, setInput] = useState("");
  const [isValid, setIsValid] = useState(true);
  const { addTask } = useContext(TodosContext);

  const submitHandler = (event: any): void => {
    event.preventDefault();
    if (!event.target.checkValidity()) {
      setIsValid(false);
    } else {
      setIsValid(true);
      addTask(+new Date(), input, () => setInput(""));
    }
  };

  return (
    <div className="row pt-2">
      <form
        className={`col needs-validation ${isValid ? "" : "was-validated"}`}
        noValidate
        onSubmit={submitHandler}
      >
        <div className="input-group mb-3">
          <input
            type="text"
            className="form-control"
            placeholder="Wprowadź zadanie"
            value={input}
            onChange={event => setInput(event.target.value)}
            maxLength={100}
            required
          />
          <div className="input-group-append">
            <button className="btn btn-primary rounded-right" type="submit">
              +
            </button>
          </div>
          <div className="valid-feedback">Pole wypełnione poprawnie</div>
          <div className="invalid-feedback">To pole nie moe być puste!</div>
        </div>
      </form>
    </div>
  );
};
