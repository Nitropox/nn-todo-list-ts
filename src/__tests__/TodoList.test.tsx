import { TodoList } from "../components/TodoList";
import React from "react";
import { shallow } from "enzyme";
import { Task } from "../context/types";

describe("TodoList component", () => {
  const mockedList: Task[] = [
    {
      date: 1570897439971,
      title: "Task 3",
      checked: false
    },
    {
      date: 1570897439972,
      title: "Task 2",
      checked: false
    },
    {
      date: 1570897439973,
      title: "Task 1",
      checked: true
    }
  ];

  it("Renders TodoList component", () => {
    expect(shallow(<TodoList list={mockedList} />).length).toEqual(1);
  });

  it("Matches the snapshot", () => {
    expect(shallow(<TodoList list={mockedList} />)).toMatchSnapshot();
  });
});
