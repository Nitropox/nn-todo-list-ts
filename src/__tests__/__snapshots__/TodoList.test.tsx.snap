// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component Matches the snapshot 1`] = `ShallowWrapper {}`;
